litestar/_signature/models/msgspec_signature_model.py

```diff
@@ -0,0 +1,132 @@
+
+    @classmethod
+    def parse_values_from_connection_kwargs(cls, connection: ASGIConnection, **kwargs: Any) -> dict[str, Any]:
+        """Extract values from the connection instance and return a dict of parsed values.
+
+        Args:
+            connection: The ASGI connection instance.
+            **kwargs: A dictionary of kwargs.
+
+        Raises:
+            ValidationException: If validation failed.
+            InternalServerException: If another exception has been raised.
+
+        Returns:
+            A dictionary of parsed values
+        """
+        try:
+            return convert(kwargs, cls, strict=False, dec_hook=dec_hook).to_dict()
+        except ValidationError as e:
+            message = str(e)
+            match = ERR_RE.search(message)
+            key = str(match.group(1)) if match else "n/a"
+            raise cls._create_exception(messages=[{"key": key, "message": message}], connection=connection) from e
+
+    def to_dict(self) -> dict[str, Any]:
+        """Normalize access to the signature model's dictionary method, because different backends use different methods
+        for this.
+
+        Returns: A dictionary of string keyed values.
+        """
+        return asdict(self)
+
+    @classmethod
+    def populate_signature_fields(cls) -> None:
+        ...
+
+    @classmethod
+    def create(
+        cls,
+        fn_name: str,
+        fn_module: str | None,
+        parsed_signature: ParsedSignature,
+        dependency_names: set[str],
+        type_overrides: dict[str, Any],
+    ) -> type[SignatureModel]:
+        struct_fields: list[tuple[str, Any, Any]] = []
+        signature_fields: dict[str, SignatureField] = {}
+
+        for parameter in parsed_signature.parameters.values():
+            annotation = type_overrides.get(parameter.name, parameter.parsed_type.annotation)
+
+            field_extra: dict[str, Any] = {"parsed_parameter": parameter}
+            meta_kwargs: dict[str, Any] = {}
+
+            if kwargs_container := parameter.kwarg_container:
+                field_extra["kwargs_model"] = kwargs_container
+                if isinstance(kwargs_container, DependencyKwarg):
+                    annotation = annotation if not kwargs_container.skip_validation else Any
+                    default = kwargs_container.default if kwargs_container.default is not Empty else NODEFAULT
+                else:
+                    param_dict = simple_asdict(kwargs_container)
+                    field_extra.update(param_dict)
+                    meta_kwargs = {
+                        k: v
+                        for k in (
+                            "gt",
+                            "ge",
+                            "lt",
+                            "le",
+                            "multiple_of",
+                            "pattern",
+                            "min_length",
+                            "max_length",
```

Comment:
No title, description, examples etc? You can see what we copy in metadata.py from annotated stuff.

Reply:
These are just the decoding constraints, nothing to do with documentation.